File: design-pattern-java/app/src/test/java/design/pattern/java/patterns/structural/adapter/ServicePrintTest.java
package design.pattern.java.patterns.structural.adapter;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class ServicePrintTest {
    @Test
    public void itShouldReturnStringWithExtendedMessageWhenServiceIsCalled() {
        String input = "some test";
        ServicePrintString service = new ServicePrintString();
        String result = service.apply(input);
        Assertions.assertNotEquals(input, result);
        Assertions.assertTrue(result.endsWith(input));
    }
    @Test
    public void itShouldReturnStringWithMessageWhenAdapterIsCalled() {
        Integer input = 345;
        AdapterIntToString serviceAdapted = new AdapterIntToString(
            new ServicePrintString());
        String result = serviceAdapted.apply(input);
        Assertions.assertNotEquals(String.valueOf(input), result);
        Assertions.assertTrue(result.endsWith(String.valueOf(input)));
    }
}

File: design-pattern-jvm/app/src/test/java/design/pattern/kotlin/patterns/structural/adapter/ServicePrintTest.kt
package design.pattern.kotlin.patterns.structural.adapter

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class ServicePrintTest {
    @Test
    fun itShouldReturnStringWithExtendedMessageWhenServiceIsCalled() {
        val input = "some test"
        val service = ServicePrintString()
        val result = service.apply(input)
        Assertions.assertNotEquals(input, result)
        Assertions.assertTrue(result.endsWith(input))
    }

    @Test
    fun itShouldReturnStringWithMessageWhenAdapterIsCalled() {
        val input = 345
        val serviceAdapted = AdapterIntToString(
            ServicePrintString()
        )
        val result = serviceAdapted.apply(input)
        Assertions.assertNotEquals(input.toString(), result)
        Assertions.assertTrue(result.endsWith(input.toString()))
    }
}